netbox/dcim/tables.py

```diff
@@ -643,41 +643,95 @@ class Meta(BaseTable.Meta):
 # Device components
 #
 
+class DeviceComponentDetailTable(BaseTable):
+    component_type = ''
+
+    pk = ToggleColumn()
+    cable = tables.LinkColumn()
+
+    class Meta(BaseTable.Meta):
+        order_by = ('device', 'name')
+        fields = ('pk', 'device', 'name', 'type', 'description', 'cable')
```

Comment:
I'd recommend removing `type` from this table and adding it only to the tables for models where it's a field (similar to how we're handling `device`).

Reply:
I have adjusted Interface and DeviceBay to instead not inherit the Component meta, as those are the only ones that will have a differing field list.